File: CLICON_TICKETERA_SOAP_JAVA_GR01/src/main/java/ec/edu/monster/model/mapper/PurchaseMapper.java
package ec.edu.monster.model.mapper;

import ec.edu.monster.model.dto.PurchaseResponse;
import ec.edu.monster.model.entity.Purchase;

public class PurchaseMapper {
    public static Purchase toEntity(PurchaseResponse dto) {
        if (dto == null)
            return null;

        return new Purchase(
                dto.idFactura(),
                dto.cedula(),
                dto.fecha(),
                dto.subtotal(),
                dto.iva(),
                dto.total());
    }
}

File: CLIESC_TICKETERA_SOAP_JAVA_GR01/src/renderer/src/mappers/Purchase.mapper.ts
import { Purchase } from '@renderer/entities/Purchase'
import { Mapper, PurchaseResponse } from '@renderer/types'

export class PurchaseMapper implements Mapper<PurchaseResponse, Purchase> {
  toEntity(dto: PurchaseResponse) {
    return new Purchase(
      dto.idFactura,
      dto.cedula,
      new Date(dto.fecha.replace(/\[.*\]$/, '')),
      dto.subtotal,
      dto.iva,
      dto.total
    )
  }
}
